app/views/handlers/submission.py

```diff
@@ -27,15 +29,21 @@ def submitted_at(self):
 
     def submit_questionnaire(self):
         payload = self.get_payload()
+
         message = json_dumps(payload)
 
         encrypted_message = encrypt(
             message, current_app.eq["key_store"], KEY_PURPOSE_SUBMISSION
         )
+
+        receipting_keys = self._metadata["receipting_keys"]
+
         submitted = current_app.eq["submitter"].send_message(
             encrypted_message,
-            case_id=self._metadata["case_id"],
-            tx_id=self._metadata.get("tx_id"),
+            case_id=self._metadata.case_id,
+            tx_id=self._metadata.tx_id,
+            receipting_keys=receipting_keys,
+            metadata=self._questionnaire_store.metadata,
         )
```

Comment:
At this point you need to generate the additional metadata. You could use a method as you need the same functionality in the flush endpoint.

Reply:
Updated 👍